### PR TITLE
Fix errors, Some features

### PR DESCRIPTION
**Pull request by IDjinn**

### CODE REVIEW

File: routes/bots.js [hunk 1]
```diff
@@ -5,104 +5,171 @@ const bot = require("../utils/discordbot");
...
-                    url: req.originalUrl
-                });
-            
-            db.Bots.findById(b.id).exec().then(async dbUser => {
-                if (dbUser)
-                    return res.render("message", {
-                        message: `O ID fornecido pertence a ${userToString(user)}, que já está cadastrado no sistema.`,
-                        url: req.originalUrl
-                    });
-                if (owners.size) {
```
Comment (IDjinn) [inline routes/bots.js:59]: Not needed....

Owners is always a list

File: routes/bots.js [hunk 2]
```diff
@@ -5,104 +5,171 @@ const bot = require("../utils/discordbot");
...
-    
-    router.post("/add", async (req, res) => {
-        if (!req.session.user) 
-            return res.redirect("/oauth2/login");
-        const b = req.body;
-        if (!((await getCaptchaResponse(config.recaptcha.secret, b["g-recaptcha-response"])).success))
-            return res.render("message", {message: "O Captcha precisa ser validado.", url: req.originalUrl});
-        const allTags = Object.keys(tags);
-        const botTags = [...new Set(typeof b.tags == "string" ? [b.tags] : b.tags)];
-        const owners = typeof b.owners == "string" ? [b.owners] : b.owners;
```
Comment (IDjinn) [inline routes/bots.js:30]: If owners field is empty, owners will be undefined, not a list... And then it will throw error

File: routes/bots.js [hunk 3]
```diff
@@ -5,104 +5,171 @@ const bot = require("../utils/discordbot");
...
+          return res.render("message", {
+            message: "Descrição longa é inválida.",
+            url: req.originalUrl,
+          });
+        if (
+          !botTags.length ||
+          botTags.length > 6 ||
+          botTags.some((t) => !allTags.includes(t))
         )
-            return res.render("message", {message: "Algo de errado não está certo.", url: req.originalUrl});
```
Comment (IDjinn) [inline routes/bots.js:41]: More specific errors to user know what happened and what didn't work

File: utils/bot.js [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
-        votes: bot.votes.current
-    }
+  return {
+    tags: bot.details.tags,
+    avatar: avatarFormat(bot),
+    name: bot.username,
+    id: bot.id || bot._id,
+    status: bot.status,
+    description: bot.details.shortDescription,
+    votes: bot.votes ? bot.votes.current : -1,
```
Comment (IDjinn) [inline utils/bot.js:11]: I've added my bot, but when i open home page, throw error in these line 🤔 
Simple fix :yey:

File: routes/bots.js (hunk 1 above)
Comment (Takasakiii) [reply routes/bots.js:59]: This was to make it compatible with the current BPD database, which is not always a list

File: routes/bots.js (hunk 3 above)
Comment (Takasakiii) [reply routes/bots.js:41]: We were doing this more specific check on the front end, but it really seems better to do both

**Review by Takasakiii — COMMENTED** [review]
Apart from the compatibility part with the old BPD, I have a lot to thank for the work helped a lot 😀

File: routes/bots.js (hunk 1 above)
Comment (IDjinn) [reply routes/bots.js:59]: But you've converted as list
Def: https://github.com/zuraaa-projects/Zuraaa.com/blob/2925fff76971d9c125625a2977a2507b46c8dce4/routes/bots.js#L30

Using: https://github.com/zuraaa-projects/Zuraaa.com/blob/2925fff76971d9c125625a2977a2507b46c8dce4/routes/bots.js#L32

😕

**Comment by IDjinn** [comment]
You're welcome ❤️

File: routes/bots.js (hunk 1 above)
Comment (Takasakiii) [reply routes/bots.js:59]: Things were so busy that I had forgotten about it, thanks for the comments

**Pull request merged**